src/main/java/me/jellysquid/mods/lithium/mixin/entity/collision_only_boat_shulker/MixinWorld.java

```diff
@@ -0,0 +1,50 @@
+package me.jellysquid.mods.lithium.mixin.entity.collision_only_boat_shulker;
+
+import com.google.common.collect.Lists;
+import me.jellysquid.mods.lithium.common.world.ExtendedChunk;
+import me.jellysquid.mods.lithium.common.world.ExtendedWorld2;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.ChunkManager;
+import net.minecraft.world.chunk.WorldChunk;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+@Mixin(World.class)
+public abstract class MixinWorld implements ExtendedWorld2 {
+
+    @Shadow
+    public abstract Profiler getProfiler();
+
+    @Shadow
+    public abstract ChunkManager getChunkManager();
+
+    //[VanillaCopy] custom: class filtered entity list together with excluding one entity
+    @Override
+    public <T> List<Entity> getEntitiesCustomType(Entity excluded, Class<? extends T> class_1, Box box_1, Predicate<? super T> predicate_1) {
```

Comment:
It doesn't appear that this needs to be done in a Mixin and could instead be moved to a static helper class.